components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java

```diff
@@ -951,6 +956,79 @@ public static SessionContext getSessionContextFromCache(String key) {
         return sessionContext;
     }
 
+    /**
+     * Retrieve session context from the session cache.
+     *
+     * @param request           HttpServletRequest.
+     * @param context           Authentication context.
+     * @param sessionContextKey Session context key.
+     * @return Session context key.
+     * @throws AuthenticationFailedException Error in triggering session expire event.
+     */
+    public static SessionContext getSessionContextFromCache(HttpServletRequest request, AuthenticationContext context
```

Comment:
Is it correct to throw the AuthenticationFailedException exception here?
Shouldn't it be some other FrameworkException?

Reply:
Fixed with 6e0671634815499c949ce71f6f8db045d7ad3366